src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double resizeCoefficient = 1.5;
+    private Object[] elementData;
+    private int numberOfObjects;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        numberOfObjects = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapasity();
+        elementData[numberOfObjects] = value;
+        numberOfObjects++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > numberOfObjects) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
+        }
+        ensureCapasity();
+        addElementInsideData(value, index);
+    }
 
+    private void addElementInsideData(T value, int index) {
```

Comment:
Read about [Ordering of class contents](https://mate-academy.github.io/style-guides/java/java.html#s3.4.2-ordering-class-contents)